hummingbot/connector/exchange/ndax/ndax_api_order_book_data_source.py

```diff
@@ -48,22 +58,27 @@ async def init_trading_pair_ids(cls, domain: Optional[str] = None):
             "OMSId": 1
         }
         async with aiohttp.ClientSession() as client:
-            async with client.get(f"{ndax_utils.rest_api_url(domain) + CONSTANTS.MARKETS_URL}",
-                                  params=params) as response:
-                if response.status == 200:
-                    resp_json: Dict[str, Any] = await response.json()
-
-                    results = {
-                        f"{instrument['Product1Symbol']}-{instrument['Product2Symbol']}": int(
-                            instrument["InstrumentId"])
-                        for instrument in resp_json
-                        if instrument["SessionStatus"] == "Running"
-                    }
+            throttler = throttler or cls._get_throttler_instance()
```

Comment:
You can simply include the default value of throttler here instead of using the following.
`throttler = throttler or cls._get_throttler_instance()`

Reply:
We have to be careful when using defaults values for parameters. Since those values are calculated when the module is loaded (and not every time the function is executed), if the default value is a mutable object, all executions of the function will be sharing the same mutable object (with all the problems that situation can produce)

Reply:
I agree with @aarmoa on the non-literal defaults in the signature, but I think @CrimsonJacket is suggesting we move the creation of the throttler to this line instead of going through a class method, which would make things a bit cleaner. My thinking was that, should we ever need to change something with how the throttler is initialized, this approach will require us changing only one place in the code